modules/grandpa/src/lib.rs

```diff
@@ -427,6 +415,29 @@ pub mod pallet {
 		)
 	}
 
+	/// Import a previously verified header to the storage.
+	///
+	/// Note this function solely takes care of updating the storage and does not verify
+	/// the validaty of such import.
+	pub(crate) fn insert_header<T: Config<I>, I: 'static>(
+		header: BridgedHeader<T, I>,
+		hash: Option<BridgedBlockHash<T, I>>,
```

Comment:
I don't think it makes sense for this to be an `Option` - it'd just force the caller to provide a `hash`